File: javatest/omnia/data/cache/MemoizedDoubleTest.java
package omnia.data.cache;

import static com.google.common.truth.Truth.assertThat;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class MemoizedDoubleTest {

  @Test
  public void memoize_didReturnSimpleDoubleMemoizer() {
    assertThat(MemoizedDouble.memoize(() -> 132.0)).isInstanceOf(SimpleDoubleMemoizer.class);
  }
}

File: tests/omnia/data/cache/MemoizedDoubleTest.kt
package omnia.data.cache

import com.google.common.truth.Truth
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class MemoizedDoubleTest {
  @Test
  fun memoize_didReturnSimpleDoubleMemoizer() {
    Truth.assertThat(MemoizedDouble.memoize { 132.0 }).isInstanceOf(SimpleDoubleMemoizer::class.java)
  }
}